src/matrix/room/Room.js

```diff
@@ -264,9 +268,14 @@ export class Room extends BaseRoom {
         }
     }
 
-    _updatePowerLevels(roomResponse) {
-        const powerLevelEvent = roomResponse.timeline.events.find(event => event.type === POWERLEVELS_EVENT_TYPE);
-        if (powerLevelEvent && this._powerLevels) {
+    _getPowerLevelsEvent(roomResponse) {
+        const isPowerlevelEvent = event => event.type === POWERLEVELS_EVENT_TYPE;
+        const powerLevelEvent = roomResponse.state?.events.find(isPowerlevelEvent) ?? roomResponse.timeline?.events.find(isPowerlevelEvent);
```

Comment:
wonder if you need `?.find` here?
Also, state events in timeline.events come after the state events in state.events, so you should first look in timeline.events and only if none found look in state.events.

Reply:
Hmm.. I don't know what I did but I did get something similar to "Sync Error: roomResponse.state is undefined". The same for roomResponse.timeline.

Reply:
Can't really reproduce anymore so I'll remove the `?` for now. Will keep testing though.

Reply:
no, I actually thought you may need an extra `?` when calling find, e.g:
```js
const powerLevelEvent = roomResponse.state?.events?.find(isPowerlevelEvent) ?? roomResponse.timeline?.events?.find(isPowerlevelEvent); 
```
but I misunderstood how `?` works and this does not seem to be needed, in node:
```
> let foo = {};
undefined
> foo.events?.bar.foo.find()
undefined
> 
```
So you initially had it correct with just a `?` after state and timeline
```js
const powerLevelEvent = roomResponse.state?.events.find(isPowerlevelEvent) ?? roomResponse.timeline?.events.find(isPowerlevelEvent); 
```